web/src/integration_url_modal.js

```diff
@@ -53,11 +55,30 @@ export function show_generate_integration_url_modal(api_key) {
             $topic_input.parent().toggleClass("hide", !checked);
         });
 
+        $show_integration_events.on("change", () => {
+            $("#integrations-event-container").toggleClass(
+                "hide",
+                !$show_integration_events.prop("checked"),
+            );
+            update_url();
+        });
+        $(document).on("change", ".integration-event", () => {
```

Comment:
I'd use the slightly more specific "#integration-events-container .integration-event" here and with the two other listeners below as well.